extensions/wikia/UserLogin/UserLoginSpecialController.class.php

```diff
@@ -602,18 +612,32 @@ public function changePassword() {
 		$this->pageHeading = wfMessage('resetpass')->escaped();
 		$this->initializeTemplate();
 
-		$this->username = $this->request->getVal( 'username', '' );
-		$this->password = $this->request->getVal( 'password', '' );
-		$this->newpassword = $this->request->getVal( 'newpassword', '' );
-		$this->retype = $this->request->getVal( 'retype', '' );
-		$this->editToken = $this->request->getVal( 'editToken', '' );
-		$this->loginToken = $this->request->getVal( 'loginToken', '' );
-		$this->returnto = $this->request->getVal( 'returnto', '' );
+		$username = $this->request->getVal( 'username', '' );
+		$password = $this->request->getVal( 'password', '' );
+		$newPassword = $this->request->getVal( 'newpassword', '' );
+		$retype = $this->request->getVal( 'retype', '' );
+		$loginToken = $this->request->getVal( 'loginToken', '' );
+		$returnto = $this->request->getVal( 'returnto', '' );
+
+		// TODO: figure out why we were setting editToken twice, once via request var and once
```

Comment:
This was/is a bug. The one from the request is what should be checked below with the `$this->wg->User->matchEditToken( $this->editToken )`, and then setting the value is used to make sure the form has the right token. Setting it twice as is being done means the check didn't do anything useful. So, it should match the behaviour of SpecialChangePassword.php from core ([1](https://github.com/Wikia/app/blob/58714681833/includes/specials/SpecialChangePassword.php#L69), [2](https://github.com/Wikia/app/blob/58714681833/includes/specials/SpecialChangePassword.php#L163).